src/universal/modules/admin/containers/Impersonate/Impersonate.js

```diff
@@ -0,0 +1,110 @@
+import React, {PropTypes, Component} from 'react';
+import {connect} from 'react-redux';
+import {cashay} from 'cashay';
+import Helmet from 'react-helmet';
+import {withRouter} from 'react-router';
+import LoadingView from 'universal/components/LoadingView/LoadingView';
+import {showError} from 'universal/modules/notifications/ducks/notifications';
+import {setAuthToken} from 'universal/redux/authDuck';
+import {reset as resetAppState} from 'universal/redux/rootDuck';
+
+const impersonateTokenQuery = `
+query {
+  user @cached(type: "User") {
+    email
+    id,
+    jwt,
+    name,
+    picture
+  }
+}`;
+
+function createImposter(userId, dispatch, router) {
+  const variables = {userId};
+  cashay.mutate('createImposterToken', {variables}).then(async ({data, error}) => {
+    if (error) {
+      // eslint-disable-next-line no-underscore-dangle
+      dispatch(showError({ title: 'Whoa there!', message: error._error }));
+      return;
+    }
+    const {email, id, jwt, name, picture} = data.createImposterToken;
+    const profile = {avatar: picture, email, id, name};
+    // Reset application state:
+    dispatch(resetAppState());
+    cashay.clear();
+    if (typeof window !== 'undefined' && typeof window.analytics !== 'undefined') {
+      // inform segment of the signout, wipe state:
+      window.analytics.reset();
+    }
+    // Assume the identity of the new user:
+    const options = {variables: {authToken: jwt}};
+    await cashay.mutate('updateUserWithAuthToken', options);
```

Comment:
@mattkrick I feel this is within spitting distance, but there is still an oddness.

When we assume the role of the user to impersonate and redirect, we're missing a few objects from cashay.entities:
 
    * `User`
    * `Presence`
 
I'm guessing this is because the 'updateUserWithAuthToken' mutation needs to be paired with the proper query so these fields end up in the cache.

Do you think that's right? How do `User` and `Presence` end up in cashay's entities in the first place?